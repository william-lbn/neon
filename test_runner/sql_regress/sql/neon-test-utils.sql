-- Test the test utils in pgxn/neon_test_utils. We don't test that
-- these actually consume resources like they should - that would be
-- tricky - but at least we check that they don't crash.

CREATE EXTENSION neon_test_utils;

select test_consume_cpu(1);

select test_consume_memory(20); -- Allocate 20 MB
select test_release_memory(5);  -- Release 5 MB
select test_release_memory();   -- Release the remaining 15 MB
