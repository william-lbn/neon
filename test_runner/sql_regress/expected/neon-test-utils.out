-- Test the test utils in pgxn/neon_test_utils. We don't test that
-- these actually consume resources like they should - that would be
-- tricky - but at least we check that they don't crash.
CREATE EXTENSION neon_test_utils;
select test_consume_cpu(1);
 test_consume_cpu 
------------------
 
(1 row)

select test_consume_memory(20); -- Allocate 20 MB
 test_consume_memory 
---------------------
 
(1 row)

select test_release_memory(5);  -- Release 5 MB
 test_release_memory 
---------------------
 
(1 row)

select test_release_memory();   -- Release the remaining 15 MB
 test_release_memory 
---------------------
 
(1 row)

